File: day15_part1_2023.rs

use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").unwrap();
    let sum: u64 = input.split(',')
                        .map(|step| {
                            let mut current_value = 0;
                            for c in step.chars() {
                                let ascii_code = c as u64;
                                current_value += ascii_code;
                                current_value *= 17;
                                current_value %= 256;
                            }
                            current_value
                        })
                        .sum();
    
    println!("{}", sum);
}

File: training_data/reflection-tuning/day15_part1_2023.py
def hash_algorithm(s):
    current_value = 0
    for char in s:
        current_value += ord(char)
        current_value *= 17
        current_value %= 256
    return current_value

# Read input from file
with open('input.txt', 'r') as file:
    initialization_sequence = file.read().strip()

# Split the sequence and apply HASH algorithm to each step
steps = initialization_sequence.split(',')
result = sum(hash_algorithm(step) for step in steps)

print(result)
